### ISSUE TITLE
Cookies on request level

### ISSUE THREAD

**Issue by alamothe** [square/okhttp#4426]
Is it possible to have a cookie store per request (as opposed to per client instance) with OkHttp? Would you consider adding this feature?

**Comment by swankjesse** [comment]
You can. Just set the "Cookie" header yourself! It'll be retained.

**Comment by alamothe** [comment]
I would need to handle response too, right? Would I use interceptors to do this? And try to reuse existing cookie jar implementation?

**Comment by swankjesse** [comment]
Yeah, you can study how the cookies are handled already to get a sense of what you'll need to do.
